### ISSUE TITLE
NPU 训练崩掉

### ISSUE THREAD

**Issue by jianzhnie** [OpenRLHF/OpenRLHF#1016]
环境：
![image](https://github.com/user-attachments/assets/a7ec6ffb-b128-443f-a40f-15c0b458d605)

数据：https://github.com/sail-sg/understand-r1-zero/tree/main/datasets/train/math_lvl3to5_8k

配置：

```shell
set -x


ray job submit --address="http://127.0.0.1:8265" \
   --runtime-env-json='{"working_dir": "/work_dir"}' \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 4 \
   --vllm_num_engines 4 \
   --vllm_tensor_parallel_size 1 \
   --vllm_gpu_memory_utilization 0.8 \
   --pretrain Qwen/Qwen2.5-7B \
   --remote_rm_url openrlhf/trainer/ppo_utils/math_reward_fn.py \
   --save_path ${work_dir} \
   --ckpt_path ${work_dir}/ckpt \
   --save_hf_ckpt \
   --max_ckpt_num 100 \
   --save_steps 10 \
   --logging_steps 1 \
   --eval_steps -1 \
   --micro_train_batch_size 4 \
   --train_batch_size 128 \
   --micro_rollout_batch_size 4 \
   --rollout_batch_size 128 \
   --n_samples_per_prompt 8 \
   --num_episodes 3 \
   --max_epochs 1 \
   --prompt_max_len 512 \
   --generate_max_len 2048 \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 1e-6 \
   --gamma 1.0 \
   --init_kl_coef 0 \
   --advantage_estimator group_norm \
   --system_prompt qwen_math_cot \
   --policy_loss_fn drgrpo \
   --eps_clip_high 0.28 \
   --correct_reward 1.0 \
   --incorrect_reward 0.0 \
   --prompt_data math_lvl3to5_8k/train.jsonl \
   --dataset_config default \
   --input_key problem \
   --label_key answer \
   --max_samples 100000 \
   --apply_chat_template \
   --normalize_reward \
   --adam_offload \
   --flash_attn \
   --gradient_checkpointing 
```

第一个Step 采样还正常， 训练的 loss 也正常

![image](https://github.com/user-attachments/assets/c5528632-4678-4476-9902-a0db261f4430)

![image](https://github.com/user-attachments/assets/20d006de-1ead-40e5-9296-810c3f136fd4)

第二个Step 采样乱码了， 训练的 loss 直接为0了

![image](https://github.com/user-attachments/assets/1f1c6003-17e9-401a-b949-65ea5676de5a)
![image](https://github.com/user-attachments/assets/4678b0af-6ad2-4e4c-bfbc-ca5d33aa383b)

_Originally posted by @jianzhnie in https://github.com/OpenRLHF/OpenRLHF/issues/605#issuecomment-2882074724_
